File: ncarg2d/src/libncarg_gks/awi/gqnt.f
C
C	$Id: gqnt.f,v 1.5 2008-07-27 00:20:59 haley Exp $
C                                                                      
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      SUBROUTINE GQNT(NTNR,ERRIND,WINDOW,VIEWPT)
C
C  INQUIRE NORMALIZATION TRANSFORMATION
C
      include 'gkscom.h'
C
      INTEGER NTNR,ERRIND
      REAL    WINDOW(4),VIEWPT(4)
C
C  Check that GKS in in the proper state.
C
      CALL GZCKST(8,-1,ERRIND)
      IF (ERRIND .NE. 0) GOTO 100
C
C  Check that the transformation number is in range.
C
      IF (NTNR.LT.0 .OR. NTNR.GT.MNT) THEN
        ERRIND = 50
        GO TO 100
      ENDIF
C
      INDX = NTNR+1
      WINDOW(1) = NTWN(INDX,1)
      WINDOW(2) = NTWN(INDX,2)
      WINDOW(3) = NTWN(INDX,3)
      WINDOW(4) = NTWN(INDX,4)
      VIEWPT(1) = NTVP(INDX,1)
      VIEWPT(2) = NTVP(INDX,2)
      VIEWPT(3) = NTVP(INDX,3)
      VIEWPT(4) = NTVP(INDX,4)
      RETURN
C
  100 CONTINUE
      WINDOW(1) = 0.
      WINDOW(2) = 0.
      WINDOW(3) = 0.
      WINDOW(4) = 0.
      VIEWPT(1) = -1.
      VIEWPT(2) = -1.
      VIEWPT(3) = -1.
      VIEWPT(4) = -1.
      RETURN
      END

File: ncarg2d/man/gks/gqnt.m
.\"
.\"	$Id: gqnt.m,v 1.16 2008-12-23 00:03:03 haley Exp $
.\"
.TH GQNT 3NCARG "March 1993" UNIX "NCAR GRAPHICS"
.SH NAME
GQNT (Inquire normalization transformation) - retrieves the 
window and viewport associated with a given normalization transformation. 
Users of NCAR Graphics are encouraged to use the SPPS routines SET
and GETSET to handle normalization transformations instead of the
GKS entries GSWN, GSVP, GSELNT, and GQNT.
.SH SYNOPSIS
CALL GQNT (NTNR, ERRIND, WINDOW, VIEWPT)
.SH C-BINDING SYNOPSIS
#include <ncarg/gks.h>
.sp
void ginq_norm_tran(Gint num, Gint *err_ind, Gtran *norm_tran);
.SH DESCRIPTION
.IP NTNR 12
(Integer, Input) - Gives the number of the normalization transformation.
.IP ERRIND 12
(Integer, Output) - If the inquired value cannot be returned correctly,
a non-zero error indicator is returned in ERRIND, otherwise a zero is returned.
Consult "User's Guide for NCAR GKS-0A Graphics" for a description of the
meaning of the error indicators.
.IP "WINDOW(4)" 12
(Real Array, Output) - The world coordinate window limits in the order
WXMIN, WXMAX, WYMIN, WYMAX.
.IP "VIEWPT(4)" 12
(Real Array, Output) - The normalized device coordinate viewport limits
in the order
VXMIN, VXMAX, VYMIN, VYMAX.
.SH ACCESS
To use GKS routines, load the NCAR GKS-0A library ncarg-gks.
.SH SEE ALSO
Online:
gswn, gsvp, gselnt, gqcntn, gqmntn, getset, gsclip, gqclip, ginq_norm_tran
.sp
Hardcopy:
"User's Guide for NCAR GKS-0A Graphics"
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br

The use of this Software is governed by a License Agreement.
